collect_app/src/androidTest/java/org/odk/collect/android/regression/FillBlankFormWithRepeatGroupTest.java

```diff
@@ -0,0 +1,197 @@
+                .pressBack(new GeneralSettingsPage(main))
+                .pressBack(new FormEntryPage("TestRepeat", main))
+                .clickForwardButton()
+                .clickOnDoNotAddGroup()
+                .clickOnDoNotAddGroup()
+                .clickForwardButton()
+                .clickSaveAndExit();
+    }
+
+    @Test
+    public void nestedGroupsWithFieldListAppearance_ShouldBeAbleToFillTheForm() {
+
+        //TestCase5
+        new MainMenuPage(main)
+                .startBlankForm("form1")
+                .swipeToNextQuestion()
+                .swipeToNextQuestion()
+                .clickSaveAndExit();
+
+        new MainMenuPage(main)
+                .startBlankForm("form2")
+                .closeSoftKeyboard()
+                .swipeToNextQuestion()
+                .clickSaveAndExit();
+
+        new MainMenuPage(main)
+                .startBlankForm("form3")
+                .closeSoftKeyboard()
+                .swipeToNextQuestion()
+                .clickSaveAndExit();
+
+        new MainMenuPage(main)
+                .startBlankForm("form4")
+                .inputText("T1")
+                .closeSoftKeyboard()
+                .swipeToNextQuestion()
+                .inputText("T2")
+                .closeSoftKeyboard()
+                .swipeToNextQuestion()
+                .inputText("T3")
+                .closeSoftKeyboard()
+                .swipeToNextQuestion()
+                .clickSaveAndExit();
+
+        new MainMenuPage(main)
+                .startBlankForm("form5")
+                .inputText("T1")
+                .closeSoftKeyboard()
+                .swipeToNextQuestion()
+                .inputText("T2")
+                .closeSoftKeyboard()
+                .swipeToNextQuestion()
+                .inputText("T3")
+                .closeSoftKeyboard()
+                .swipeToNextQuestion()
+                .clickSaveAndExit();
+
+        new MainMenuPage(main)
+                .startBlankForm("form6")
+                .inputText("T1")
+                .closeSoftKeyboard()
+                .swipeToNextQuestion()
+                .inputText("T2")
+                .closeSoftKeyboard()
+                .swipeToNextQuestion()
+                .inputText("T3")
+                .closeSoftKeyboard()
+                .swipeToNextQuestion()
+                .clickSaveAndExit();
+
+        new MainMenuPage(main)
+                .startBlankForm("form7")
+                .swipeToNextQuestion()
+                .clickSaveAndExit();
+
+        new MainMenuPage(main)
+                .startBlankForm("form8")
+                .closeSoftKeyboard()
+                .swipeToNextQuestion()
+                .clickSaveAndExit();
+
+        new MainMenuPage(main)
+                .startBlankForm("form9")
+                .closeSoftKeyboard()
+                .swipeToNextQuestion()
+                .clickSaveAndExit();
+    }
+
+    @Test
+    public void repeatGroupNotAdded_ShouldBackButtonBeActive() {
```

Comment:
Again not sure I understand this test name. What are we testing for here?

Reply:
I changed the name to 'ShouldBackwardButtonBeClickable'
We had a bug where the backward button was inactive and unclickable when no group was added.

Reply:
I think as in the above this could be `whenNoRepeatGroupAdded_backButtonIsClickable`.

Reply:
As with the above I think this could be `whenNoRepeatGroupAdded_backButtonIsClickable`.